src/custom-element.js

```diff
@@ -1092,6 +1092,14 @@ function createBaseCustomElementClass(win) {
       return waitFor.then(() => this.implementation_);
     }
 
+    /**
+     * Returns reference to upgraded implementation, where imperative APIs are placed.
+     * @return {!Promise<!./base-element.BaseElement>}
+     */
+    getApi() {
```

Comment:
Do we need this method? And if we do, we should probably return `this.getImpl().getApi()`? And maybe even only after `whenUpgraded`?

Reply:
Ok. Re-reviewing again. IIUC your intention here was to do:

```
customElements.whenDefined(element.tagName).then(() => element.getApi())
```

?

If so, this makes sense to me. We can always dress it up a bit more in additional utilities exported via `AMP` object or via a statically compiled npm package utility.  But otherwise, this addition here makes sense. Again, assuming this should be `this.getImpl().getApi()`.